arangod/Cluster/ClusterInfo.cpp

```diff
@@ -224,6 +230,51 @@ void ClusterInfo::cleanup() {
   theInstance->_currentCollections.clear();
 }
 
+
+void ClusterInfo::triggerBackgroundGetIds() {
+  // Trigger a new load of batches
+  _uniqid._nextBatchStart = 1ULL;
+  _uniqid._nextUpperValue = 0ULL;
+
+  LOG_DEVEL << "Trigger background get ids";
+
+  try {
+    if (_uniqid._backgroundJobIsRunning) {
+      return ;
+    }
+    _uniqid._backgroundJobIsRunning = true;
+    std::thread([this] {
+      uint64_t result;
+      try {
+        result = _agency.uniqid(MinIdsPerBatch, 0.0);
+      } catch (std::exception const& e) {
+        _uniqid._backgroundJobIsRunning = false;
+        return ;
+      }
+
+      {
+        MUTEX_LOCKER(mutexLocker, _idLock);
+
+        //LOG_DEVEL << "background got ids: " << result
+        //  << " _nextBatchStart is " << _uniqid._nextBatchStart;
+
+        if (1ULL == _uniqid._nextBatchStart) {
+          // Invalidate next batch
+          _uniqid._nextBatchStart = result;
+          _uniqid._nextUpperValue = result + MinIdsPerBatch - 1;
+          //LOG_DEVEL << "Updated next batch";
+        } else {
+          //LOG_DEVEL << "Throw away ids, someone was faster";
+        }
+        // If we get here, somebody else tried succeeded in doing the same,
+        // so we just try again.
+      }
```

Comment:
am I missing something or is this thread not resetting the `_backgroundJobIsRunning` in positive result